enumerables.rb

```diff
@@ -0,0 +1,184 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/BlockNesting
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield self[i]
+        else
+          yield(keys[i], values[i])
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < length
+        if instance_of?(Array)
+          yield(self[i], i)
+        else
+          yield(values[i], i)
+        end
+        i += 1
+      end
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_select
+    if block_given?
+      i = 0
+      new_arr = []
+      new_hash = {}
+      while i < length
+        if instance_of?(Array)
+          new_arr << self[i] if yield self[i]
+        elsif yield(keys[i], values[i])
+          new_hash[keys[i]] = values[i]
+        end
+        i += 1
+      end
+
+      return new_arr if instance_of?(Array)
+      return new_hash if instance_of?(Hash)
+
+    else
+      puts 'block was not given'
+    end
+  end
+
+  def my_all?(sth = nil)
+    return true if empty?
+
+    if block_given?
+      my_each do |item|
+        return false unless yield item
+      end
+    else
+      my_each do |item|
+        if sth.instance_of?(Regexp)
+          return false if item.match?(sth) == false
+        elsif sth.is_a? Class
+          return false unless item.is_a? sth
+        elsif [false, nil].include?(item)
+          return false
+        end
+      end
+    end
+    true
+  end
+
```

Comment:
- [ ]  enumerables #my_all? returns false if the block returns false or nil::range
      ``` Failure/Error: expect(range.my_all?(&false_block)).to eq(range.all?(&false_block)) ```      
      ``` NoMethodError: undefined method `empty?' for 5..50:Range ```

- [ ]  enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
      ``` Failure/Error: expect(array.my_all?(3)).to be array.all?(3) ```
      ``` expected false but got true ```